### DISCUSSION TITLE
How to use msgpack with socket.io client

### DISCUSSION THREAD

**Discussion by ghost** [socketio/socket.io#4231] [Q&A] [answered]
Hello, i'm trying to use msgpack with socket.io.

in the doc, they say that :

![unknown](https://user-images.githubusercontent.com/86203108/147817220-5a3f86f7-ce92-4f77-aa41-1ede08f0746b.png)
,

So I made that :

```js
//Server
import { Server } from "socket.io";

import { createServer } from "http";

import customParser from "socket.io-msgpack-parser";

const httpServer = createServer();

const io = new Server(httpServer, {
    parser: customParser,
});

httpServer.listen(8080);



//Client

//HTML
<script src="https://cdnjs.cloudflare.com/ajax/libs/socket.io/4.4.0/socket.io.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/socket.io/4.4.0/socket.io.msgpack.min.js"></script>

//JS
const socket = io("ws://localhost:8080");
```

But it's actually not working and give this error 
![unknown](https://user-images.githubusercontent.com/86203108/147817304-0c87f761-a1c7-4be3-aa59-46a614b9e9aa.png)
:

If i remove the parser option on the server, it's work fine. So how can I use msgpack on socket.io client ?

**Comment by darrachequesne** [comment, marked as answer]
Hi! The `socket.io.msgpack.min.js` bundle already contains the Socket.IO client, so you don't need to include `socket.io.min.js`.

**Reply by ghost** [reply]
Oh ok, thanks
